indico/modules/rb_new/client/js/components/TimelineItem.jsx

```diff
@@ -70,26 +73,53 @@ export default class TimelineItem extends React.Component {
     };
 
     renderOccurrence = (occurrence, additionalClasses = '') => {
-        const {start_dt: startDt, end_dt: endDt, reservation, bookable} = occurrence;
+        let segmentStartDt, segmentEndDt, popupContent;
+        const {start_dt: startDt, end_dt: endDt, start_time: startTime, end_time: endTime,
+               reservation, reason, bookable} = occurrence;
         const {startHour, endHour, step, onClick} = this.props;
-        const segmentStartDt = moment(startDt, 'YYYY-MM-DD HH:mm');
-        const segmentEndDt = moment(endDt, 'YYYY-MM-DD HH:mm');
+        if (additionalClasses === 'blocking') {
```

Comment:
I'd rather add a new `type` argument instead of abusing the classes for this (the name already implies that there can be other classes in it as well, which would break this)